addons/partner_autocomplete/models/res_company.py

```diff
@@ -12,3 +20,54 @@ class ResCompany(models.Model):
     def _inverse_partner_gid(self):
         for company in self:
             company.partner_id.partner_gid = company.partner_gid
+
+    def _enrich(self):
+        """
+        This method calls the partner autocomplete service from IAP to enrich the partner related fields
+        of the company.
+        """
+        self.ensure_one()
+
+        company_domain = self._get_company_domain()
+        if not company_domain:
+            return {'error': _("Could not extract company domain. Impossible to enrich the company.")}
+        company_data = self.env['res.partner'].enrich_company(company_domain, False, self.vat)
+        if company_data.get('error'):
+            return {'error': company_data.get('error_message', _('Unkown reason'))}
+
+        additional_data = company_data.pop('additional_info', False)
+
+        if company_data.get('state_id'):
+            company_data['state_id'] = company_data.get('state_id', {}).get('id', False)
+        if company_data.get('country_id'):
+            company_data['country_id'] = company_data.get('country_id', {}).get('id', False)
+
+        # Keep only not null value and value not already set on the target partner.
+        company_data = {field: value for field, value in company_data.items()
+                        if field in self.partner_id._fields and value and not self.partner_id[field]}
+
+        self.partner_id.write(company_data)
+        if additional_data:
```

Comment:
Could move the additional_data variable closer to this line since it's not used before.